gdx/src/com/badlogic/gdx/Input.java

```diff
@@ -790,21 +790,45 @@ public enum OnscreenKeyboardType {
 		Default, NumberPad, PhonePad, Email, Password, URI
 	}
 
-	/** Vibrates for the given amount of time. Note that you'll need the permission
+	/** Generates a simple haptic effect of a given duration or a vibration effect on devices without haptic capabilities. Note
+	 * that on Android backend you'll need the permission
 	 * <code> <uses-permission android:name="android.permission.VIBRATE" /></code> in your manifest file in order for this to work.
+	 * On iOS backend you'll need to set <code>useHaptics = true</code> for devices with haptics capabilities to use them.
 	 * 
 	 * @param milliseconds the number of milliseconds to vibrate. */
 	public void vibrate (int milliseconds);
 
-	/** Vibrate with a given pattern. Pass in an array of ints that are the times at which to turn on or off the vibrator. The
-	 * first one is how long to wait before turning it on, and then after that it alternates. If you want to repeat, pass the index
-	 * into the pattern at which to start the repeat.
-	 * @param pattern an array of longs of times to turn the vibrator on or off.
-	 * @param repeat the index into pattern at which to repeat, or -1 if you don't want to repeat. */
-	public void vibrate (long[] pattern, int repeat);
+	/** Generates a simple haptic effect of a given duration and default amplitude. Note that on Android backend you'll need the
+	 * permission <code> <uses-permission android:name="android.permission.VIBRATE" /></code> in your manifest file in order for
+	 * this to work. On iOS backend you'll need to set <code>useHaptics = true</code> for devices with haptics capabilities to use
+	 * them.
+	 *
+	 * @param milliseconds the duration of the haptics effect
+	 * @param fallback whether to use non-haptic vibrator on devices without haptics capabilities (or haptics disabled) */
+	public void vibrate (int milliseconds, boolean fallback);
```

Comment:
I've added a new line warning that amplitude and length parameters are ignored on fallback (non-haptic) vibrations. The 50ms is Android specific (iOS non-haptic supported vibration sound doesn't allow configuring length) and I don't think it makes sense to be able to configure it on Android. To be honest, setting `fallback` to `true` should be very rare, in most cases, you'd want to only make a device vibrate if it has proper haptics support.

Reply:
I would not say so. In my game, I use different vibration lengths at the moment (on Android) for different intensities. 

https://github.com/MrStahlfelge/lightblocks/blob/848ed974d2e436350f793333347f3872e41594f1/core/src/de/golfgl/lightblocks/input/VibrationType.java#L13

As you can see, I use 20 for haptic feedback which seems to be okay for the players. I think 50 is a bit too much, but of course this depends on the given vibrationType.

Reply:
I've been thinking about this abit more and probably the best approach is to just remove the `fallback` on this particular method. The whole point about this method is that it provides presets for different haptic intensities and tries to ensure they will feel ok on all devices no matter the OS or device model (taking advantage of the new API that lets manufacturers to adjust amplitude and length for each particular device). On the other hand, the `fallback` vibrations on non-haptic devices will be a one shot vibration with the same intensity no matter the passed `VibrationType` and will feel too short or too long or too intense or not depending on the device and use case. Not being able to provide a good enough `fallback` best is to remove it.

To support old devices we'll still be able to use `vibrate ()` methods that receive `legth` as an argument.

Reply:
I think it is better not to use a fallback, but client code should have a way to know if something actually happened to switch to alternatives. Returning a success boolean should be good

Reply:
I agree there may be the need to know if the device supports haptics or not but this info should be available beforehand, without having to make a call to the method. This may also be useful to only display a settings button to enable/disable vibration on devices that support haptics. Also, there's no guarantee (at least on iOS) that a haptic feedback occurs as it may be overriden by higher priority events by the OS.

Currently you can check for vibrator support using `isPeripheralAvailable(Peripheral)` but it doesn't distinguish between haptic and non-haptic vibrators.
I think the best solution would be to add a new `Peripheral` called `Haptics` or `HapticFeedback` or something like that. What do you think?

Reply:
This is even better, my suggestion was to keep the effort low.